api/src/main/java/org/infinispan/api/Infinispan.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.api;
+
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.api.reactive.KeyValueStore;
+import org.infinispan.api.reactive.KeyValueStoreConfig;
+
+/**
+ * Each infinispan is a node instance. It can be embedded or client node, depending on the access point. {@link
+ * InfinispanClient} or {@link InfinispanEmbedded}
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+@Experimental
+public interface Infinispan {
+
+   /**
+    * Gets the {@link KeyValueStore} by name.
+    * <p>
+    * If the store does not exist, creates a {@link KeyValueStore} with the default values
+    *
+    * @param name, name of the store
+    * @return {@link KeyValueStore}
+    */
+   <K, V> KeyValueStore<K, V> getKeyValueStore(String name);
+
+   /**
+    * Gets the {@link KeyValueStore} by name.
+    * <p>
+    * If the store does not exist, creates a {@link KeyValueStore} with the given config
+    *
+    * @param name, name of the store
+    * @return {@link KeyValueStore}
+    */
+   <K, V> KeyValueStore<K, V> getKeyValueStore(String name, KeyValueStoreConfig config);
```

Comment:
This might be a pet hatred of mine, but I dislike more and more overloaded methods. We have the experience of `getCache()` vs `getCache(String)` to compare as well as many others. I'd favour a single method and an easy to find static method in `KeyValueStoreConfig` that returns the default configuration.

Reply:
ok. in any case that can be done and if we decide we want to add the one without the default we can do it later.